### DISCUSSION TITLE
Opinions on squashing PR commits

### DISCUSSION THREAD

**Discussion by msujew** [eclipse-langium/langium#177] [General]
Hi team,

I was wondering whether we should start squashing our PR commits. Assuming Langium will go on for a few years, we will accumulate a lot of commits over time. As Git performance suffers under a large amount of commits, I would argue in favor of squashing commits. What do others think?

**Comment by kaisalmen** [comment]
I would not enforce it, but suggest something like a review step for yourself when opening a PR. There are situation where squash-it-all-no-matter-what is not the most meaningful approach IMO.
